src/Ocelot/DependencyInjection/ServiceCollectionExtensions.cs

```diff
@@ -47,7 +51,7 @@ public static IOcelotBuilder AddOcelot(this IServiceCollection services, IConfig
     /// <returns>An <see cref="IOcelotBuilder"/> object.</returns>
     public static IOcelotBuilder AddOcelotUsingBuilder(this IServiceCollection services, Func<IMvcCoreBuilder, Assembly, IMvcCoreBuilder> customBuilder)
     {
-        var configuration = services.BuildServiceProvider().GetRequiredService<IConfiguration>();
+        var configuration = services.FindConfiguration(null);
         return new OcelotBuilder(services, configuration, customBuilder);
     }
```

Comment:
My suggestion is removing this version at all because of absence of `IConfiguration configuration` parameter.
It will be breaking change for sure, but we will mention it in Release Notes.

Reply:
In my opinion, this method can be removed/obsoleted but it has more to do with the custom mvc builder.

Reply:
obsolete for me, not removed... In a further step remove

Reply:
🆗 Let's make it Obsolete.

Reply:
Fixed in https://github.com/ThreeMammals/Ocelot/pull/1986/commits/293c1d9d5b5bd30b7ebcfadfc50c71b69487118b ✔️